articles/240624_gren_04.md

```diff
@@ -0,0 +1,80 @@
+
+* `FileSystem` contains functions for working with the file system.
+* `FileSystem.FileHandle` allows you to open a file, perform several operations on it and then close it again.
+* `FileSystem.Path` gives you a staticly typed, cross-platform, representation of a file system path.
+
+Together, these modules gives you access to most of the file system operations available in NodeJS.
+
+If you want more information about the new FileSystem API, be sure to watch the video linked above.
+
+## Gren-in-Gren
+
+This release contains the first few lines of what will eventually be a ground-up rewrite of the compiler in Gren. While
+it currently does little more than downloading and executing the Haskell-based compiler, it will eventually enable the
+use of compiler functionality (parsing, type checking, formatting) in Gren applications.
+
+There are also several indirect benefits of the Gren-in-Gren rewrite, like lowering the barrier of contributing to the
+compiler, and improving the quality of the Gren core packages by virtue of compiler authors now also being Gren-developers.
+
+In fact, the mentioned rewrite of the FileSystem API is a direct consequence of the compiler needing more functionality
+than what was previously available.
+
+We'll talk more about the importance of Gren-in-Gren at a later time.
+
+## The book has received a face lift
+
+Justin Blake has contributed big changes to [the book](https://gren-lang.org/book), which serves as our language guide.
+
+Not only has he filled in empty chapters and added a few new ones, he's also replaced the generator with Astro Starlight.
+One of the biggest consequences of this change is that you can now search for topics of interest.
+
+## We've moved to Discord
+
+Since the very first release of Gren we've been using Zulip to coordinate development of the language. Over time we've received feedback
+that Zulip might not have been the best choice. Many people feel that they're drowning in communication platforms, so the bar for joining
+yet another platform is pretty high.
+
+While our experience with Zulip has been good, the future success of Gren depends on an active community.
+Using the best tool means nothing if we're using it alone.
+
+We've now decided to use [Discord](https://discord.gg/J8aaGMfz) for discussing and sharing all things related to Gren. Why don't you join us?
+
+## Misc. changes
+
+* Using multi-line string syntax for a single-line string is now a compile error.
+* If the `NO_COLOR` environment variable is set, color will be stripped from the compiler output.
+* `Stream.sendX` and `Node.exit` functions are now `Task`-based. This makes them easier to compose with other `Task`s.
+* You now have access to the location of the current executing program.
+* You can now set the title of the currently running process.
+* Removed `always`, use anonymous functions instead.
```

Comment:
While an anonymous function can be nicer than usual, always has a different performance characteristic than an anonymous function, in that the computed data is only computed once, instead of multiple times.

`List.map (always (expensiveComputation x)) list` will for instance be much faster than `List.map (\_ -> expensiveComputation x) list`.
I'm not saying it's always used in the best way, but it does have it's uses.

Thankfully, people can redefine it if needed. And maybe that's good enough 👍